app/models/supported_ssg.rb

```diff
@@ -17,31 +17,35 @@ def ref_id
 
   class << self
     def supported?(ssg_version:, os_major_version:, os_minor_version:)
-      ssg_version == ssg_for_os(os_major_version, os_minor_version)
+      supported_versions = ssg_versions_for_os(os_major_version, os_minor_version)
+      supported_versions.include?(ssg_version)
     end
 
-    def ssg_for_os(os_major_version, os_minor_version)
-      raw_supported.dig(
-        'supported',
-        "#{self::OS_NAME}-#{os_major_version}.#{os_minor_version}",
-        'version'
-      )
+    def ssg_versions_for_os(os_major_version, os_minor_version)
+      os_major_version = os_major_version.to_s
+      os_minor_version = os_minor_version.to_s
+
+      all.select do |ssg|
+        ssg.os_major_version == os_major_version &&
+          ssg.os_minor_version == os_minor_version
+      end.map(&:version)
```

Comment:
I really, really wish we could keep from searching through the list like this. I'd much rather have some unique key we can use for lookups, but I can't think of a better solution at the moment :/

Reply:
The list isn't that big to search on. And this way it fits into other usage with parsed data. Hashing also takes some cycles, but I get your point.